### ISSUE TITLE
How do I use this library with esqueleto?

### ISSUE THREAD

**Issue by roosemberth** [brandonchinn178/persistent-mtl#37]
[esqueleto](https://hackage.haskell.org/package/esqueleto) is an EDSL oftentimes run on top of persistent.

Given the [grammar for `MonadSqlQuery`](https://github.com/brandonchinn178/persistent-mtl/blob/master/src/Database/Persist/Monad/SqlQueryRep.hs#L50) and [it's interpreter](https://github.com/brandonchinn178/persistent-mtl/blob/master/src/Database/Persist/Monad/SqlQueryRep.hs#L559), I fear the lib does not expose the necessary tools for lifting esqueleto queries. Did I overlook something? If I did, the question in the title holds, otherwise, would you be willing to add shims for esqueleto aswell?

**Comment by brandonchinn178** [comment]
No, you didn't overlook anything. I think this feature request makes sense, although I would hesitate to add a dependency on `esqueleto` to this library. What I'll do instead is add a feature to lift an arbitrary `SqlPersistT` action, and in your own code, you can define the `SqlQueryRep` value yourself, something like

```hs
select :: MonadSqlQuery m => E.SqlQuery a -> m [a]
select = unsafeLiftSql "esqueleto-select" . E.select
```

Would that work?

**Comment by roosemberth** [comment]
It feels like it breaks abstraction, but I think it would be a useful manhole. That would work, thanks!

NB: This is somewhat similar to how I patched it myself.

**Comment by brandonchinn178** [comment]
Released as persistent-mtl 0.3.0.0!
